dist/config.d/10-agent.toml

```diff
@@ -0,0 +1,5 @@
+# Configure agent timing.
+[agent.timing]
+
+# Pausing interval between updates checks in steady mode, in seconds.
+steady_interval_secs = 300
```

Comment:
Aside: hmm, any reason why we're shipping defaults in separate fragments vs. just a single one called e.g. `01-defaults.toml`? Is there a method to the priorities those separate fragments have (I see we're using 10, 30, and 50)?

Related to this, I wonder if we should do something similar to systemd where we just have the defaults commented out, so that it also serves as documentation on all the settings that exist.

Anyway, not blocking this of course!

Reply:
I think you are basically asking for a single file with no real content, but only docstring and commented default values.

The approach we took so far is to have explicit defaults set via config. Those are split in smaller files so that it's easier to completely discard specific sections by placing a null-symlink with the same filename.

Priorities are somewhat arbitraries, but I'm using the `[10, 50]` range to make it easier to override by a user.

Reply:
> I think you are basically asking for a single file with no real content, but only docstring and commented default values.

Yes, exactly. See e.g. https://github.com/systemd/systemd/blob/d900701eeab4a43b6626256226c68ed00a63962d/src/journal/journald.conf.

> The approach we took so far is to have explicit defaults set via config. Those are split in smaller files so that it's easier to completely discard specific sections by placing a null-symlink with the same filename.

I think the thing that seems confusing to me is that we're essentially shipping two sets of defaults: compiled-in options, and these fragments. (Though I think only the non-string ones actually have compiled-in defaults and the rest Zincati would error out on if the config is missing?). What does it mean to "mask" a default fragment? We'd just fall back to the compiled default, no? I can see the value of masking though if there are multiple layered fragments injected by other apps/layers that you want to cancel out.

Re. priorities, I think it'd be clearer if they were all e.g. `01`. Otherwise, we're just making the range under the priority we choose useless for that specific knob, no?

(Sorry, probably should split this into a separate issue for discussion!)

Reply:
Yes, we can split to another ticket.

I'd say the approach so far has been more similar to https://github.com/systemd/systemd/tree/d900701eeab4a43b6626256226c68ed00a63962d/sysctl.d instead.

We leave a bit of space before (<10) for out-of-repo defaults, and most of the range (>50) for overrides.

As you noted, some in-code defaults are not the same as the config, and just fail validation if the fragment is masked.